### PR TITLE
Refactor website, set up new assignment

### PR DESCRIPTION
**Pull request by hui444**
* Refactor website to use specific route types
* Abstract PATH const
* Refactor shopping list code
* Fix component definition missing display name errors
* Set up frontendmentor assignment
* Add shopping list hooks example

### CODE REVIEW

File: src/components/AddItemModal.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react'
+
```
Comment (limwenfeng) [inline src/components/AddItemModal.tsx]: what is the component for?

File: src/components/Pair2/ItemAdder.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React, { useState } from 'react'
 import { Input } from 'antd'
-import { ItemAdderContainer } from '../../pages/Pair2/styles/Pair2.styled'
+import { ItemAdderContainer } from '../../pages/ShoppingLists/Pair2/styles/Pair2.styled'
```
Comment (limwenfeng) [inline src/components/Pair2/ItemAdder.tsx:3]: why got Pair2 one?

Comment (limwenfeng) [reply src/components/Pair2/ItemAdder.tsx:3]: or u just helping them refactor the code

File: src/pages/Groups/GroupsSwitch.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      } else if (screenNumber === 2) {
+        return <Group4Screen2 />
+      } else {
+        return <Group4 />
+      }
+    default:
+      return <NotFound />
+  }
+}
+
```
Comment (limwenfeng) [inline src/pages/Groups/GroupsSwitch.tsx:70]: you just copy pasting this code from another file right

File: src/routes/RouteTypes.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      render={(props) => (
+        <StyledMain hasFooter>
+          <MainNavigation />
+          <Component {...props} />
+          <Footer />
+        </StyledMain>
+      )}
+    />
+  )
+}
```
Comment (limwenfeng) [inline src/routes/RouteTypes.tsx:42]: is this new or refactored?

File: src/components/AddItemModal.tsx (hunk 1 above)
Comment (hui444) [reply src/components/AddItemModal.tsx]: i added my example shopping list, this is the add item modal for it

File: src/components/Pair2/ItemAdder.tsx (hunk 2 above)
Comment (hui444) [reply src/components/Pair2/ItemAdder.tsx:3]: ye bc i moved it into a shopping list folder

File: src/pages/Groups/GroupsSwitch.tsx (hunk 3 above)
Comment (hui444) [reply src/pages/Groups/GroupsSwitch.tsx:70]: yup

File: src/routes/RouteTypes.tsx (hunk 4 above)
Comment (hui444) [reply src/routes/RouteTypes.tsx:42]: new. it's to make the routes file easier to read and also to make it more similar to rhapp code

**Comment by limwenfeng** [comment]
LGTM!

**Review by limwenfeng — APPROVED** [review]

**Pull request merged**